pytorch_optimizer/optimizer/soap.py

```diff
@@ -81,8 +81,8 @@ def reset(self):
             for p in group['params']:
                 state = self.state[p]
 
-                state['exp_avg'] = torch.zeros_like(p)
-                state['exp_avg_sq'] = torch.zeros_like(p)
+                state['exp_avg'] = torch.zeros_like(p, dtype=p.dtype)
+                state['exp_avg_sq'] = torch.zeros_like(p, dtype=p.dtype)
```

Comment:
```suggestion
                state['exp_avg'] = torch.zeros_like(p)
                state['exp_avg_sq'] = torch.zeros_like(p)
```

`torch.zeros_like(p)` is equivalent to `torch.zeros(p.size(), dtype=p.dtype, layout=p.layout, device=p.device)`, so I think we don't need to explicit `dtype` here!